File: Device/Device.cpp
/*
 * Device.cpp
 *
 *  Created on: Jun 18, 2012
 *      Author: jmonk
 */

#include "Device.h"
#include "../Comm/Communicator.h"
#include "../Packets/Packet.h"

namespace dvs {

map<int, Device*> devicesById;
map<string, Device*> devicesByString;

Device::Device(Communicator* comm, string name) {
	deviceId = Device::allocateDeviceId();
	Device::setDevice(deviceId, name, this);
	this->name = name;
	this->comm = comm;
	this->cname = "Unset";
	roomId = -1;
}

Device::~Device() {

}

unsigned int Device::getId() {
	return deviceId;
}

string Device::getName() {
	return name;
}

string Device::getCName() {
	return cname;
}

void Device::setCName(string cname) {
	this->cname = cname;
}

void Device::send(Packet* packet) {
	comm->sendPacket(packet);
}

Room* Device::getRoom() {
	if (roomId >= 0) {
		return Server::getRoom(roomId);
	} else {
		return NULL;
	}
}

void Device::setRoom(unsigned int roomId) {
	if (this->roomId >= 0) {
		Server::getRoom(this->roomId)->remDevice(this);
	}
	this->roomId = roomId;
	if (this->roomId >= 0) {
		Server::getRoom(this->roomId)->addDevice(this);
	}
}


void Device::setLoc(float x, float y) {
	this->x = x;
	this->y = y;
}

void Device::setIcon(unsigned int icon) {
	this->icon = icon;
}

bool Device::hasField(unsigned char id) {
	return fields.end() != fields.find(id);
}

Field* Device::getField(unsigned char id) {
	return fields[id];
}

void Device::setField(unsigned char id, Field* field) {
	if (hasField(id)) {
		delete fields[id];
	}
	fields[id] = field;
}

} /* namespace DVS */

File: JavaTestClient/src/com/dreamcrushed/DeviceUser/Device.java
package com.dreamcrushed.DeviceUser;

public class Device {

	private int id;
	private String name;
	private String cname;
	private int room;
	private int icon;
	private float x;
	private float y;

	public Device(int id, String name, String cname, int room, int icon,
			float x, float y) {
		this.id = id;
		this.name = name;
		this.cname = cname;
		this.room = room;
		this.icon = icon;
		this.x = x;
		this.y = y;
		System.out.println("Created Device: " + name);
		System.out.println("Common Name: " + cname);
		System.out.println("Id: " + id + " Room: " + room + " X-Y:" + x + "-" + y + "\n");
	}

	public void setField(int id, String name, FieldType type, boolean writable,
			boolean vol) {
		System.out.println("Field Defined for " + this.name);
		System.out.println("Id: " + id + "\t Name: " + name);
		System.out.println("Type: " + type + " Writable: " + writable + " Volatile: " + vol + "\n");
	}

}
